src/Core/Form/IdentifiableObject/DataProvider/SpecificPriceFormDataProvider.php

```diff
@@ -67,6 +75,9 @@ public function getData($id): array
                 'currency_id' => $specificPriceForEditing->getCurrencyId(),
                 'country_id' => $specificPriceForEditing->getCountryId(),
                 'group_id' => $specificPriceForEditing->getGroupId(),
+                //@todo: this is also used to get combination choices based on shop, but probably it should be dynamic -
+                //       if all shops are chosen, then combination from all shops should be loaded to choice list,
+                //       if one shop is selected, then that shop combinations list should be loaded.
```

Comment:
Probably combinationIdChoices and shopId choices must dynamically update depending on each other.
Because in All shops context I should be able to apply spec.price for any combination in any shop and those combinations might be different in shops. So contextShop combinations != all shops combinations. So I think I should handle this in a dedicated PR via javascript :thinking: 
wdyt @jolelievre?

Reply:
Yep I agree this should be adapted, but it's not really related with this PR Since Mateus is gonna create a dedicated issue for this dynamic content I think you can remove this comment here